src/components/Dropdown.tsx

```diff
@@ -0,0 +1,282 @@
+    case ActionType.ToggleDropdown: {
+      return {
+        ...state,
+        isDropdownOpen: !state.isDropdownOpen,
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+/*------------------------------------------------------------------------*/
+/* ------------------------------ Component ----------------------------- */
+/*------------------------------------------------------------------------*/
+
+const Dropdown: React.FC<Props> = (props) => {
+  /*------------------------------------------------------------------------*/
+  /* -------------------------------- Setup ------------------------------- */
+  /*------------------------------------------------------------------------*/
+
+  /* -------------- Props ------------- */
+
+  // Destructure all props
+  const {
+    selectedItem,
+    items,
+    onChange,
+    variant,
+    onlyRenderContentInButton,
+  } = props;
+
+  /* -------------- State ------------- */
+
+  // Initial state
+  const initialState: State = {
+    isDropdownOpen: false,
+  };
+
+  // Initialize state
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  // Destructure common state
+  const {
+    isDropdownOpen,
+  } = state;
+
+  /*----------------------------------------*/
+  /* --------------- Main UI -------------- */
+  /*----------------------------------------*/
+
+  return (
+    <div className="dropdown">
+      <button
+        className={`btn dropdown-toggle border 
+          ${isDropdownOpen && 'show'}
+          ${variant && variant === Variant.Dark && 'btn-dark text-light'} 
+          ${variant && variant === Variant.Light && 'btn-light text-dark'}
+        `}
+        type="button"
+        id="dropdownMenuButton"
+        aria-expanded={isDropdownOpen}
+        aria-label={`${selectedItem.label} currently selected. Click to change`}
+        onClick={() => {
+          dispatch({
+            type: ActionType.ToggleDropdown,
+          });
+        }}
+      >
+        {
+          selectedItem.content
+            ? selectedItem.content
+            : selectedItem.label
+        }
+      </button>
+      <ul
+        className={`dropdown-menu 
+          ${variant && variant === Variant.Dark ? 'dropdown-menu-dark' : ''} 
+          ${isDropdownOpen && 'show'}
+        `}
+        aria-labelledby="dropdownMenuButton"
+      >
+        {Object.values(items).map((item) => {
+          return (
+            <li
+              key={item.label}
+            >
+              <button
+                type="button"
+                aria-label={`Show ${item.label}`}
+                className={`dropdown-item ${selectedItem.label === item.label
+                  ? 'active-custom'
+                  : ''
+                } 
+                `}
+                onClick={(e) => {
+                  e.preventDefault();
+                  onChange(item);
+                  dispatch({
+                    type: ActionType.ToggleDropdown,
+                  });
```

Comment:
I'm worried about what happens when the user clicks _outside_ the dropdown. Does it go away?